pywbem/cim_obj.py

```diff
@@ -3880,6 +3880,17 @@ def __init__(self, name, type, value=None, is_array=False,
         self.tosubclass = tosubclass
         self.toinstance = toinstance
         self.translatable = translatable
+        if is_array is None:
+            raise TypeError('None not allowed for is_array')
+        if array_size and not is_array:
+            raise TypeError('QualifierDeclaration array_size allowed only '
+                            'when is_array True.')
+        if isinstance(value, list) and not is_array:
+            raise TypeError('QualifierDeclaration is_array False and '
```

Comment:
Think the difference is that you and I view Value vs Type a little differently. All are changed as is the test_cim_obj.py